server/src/main/java/io/druid/client/cache/MemcachedCache.java

```diff
@@ -79,35 +136,174 @@ public static MemcachedCache create(final MemcachedCacheConfig config)
+                  .setProtocol(ConnectionFactoryBuilder.Protocol.BINARY)
+                  .setLocatorType(ConnectionFactoryBuilder.Locator.CONSISTENT)
+                  .setDaemon(true)
+                  .setFailureMode(FailureMode.Cancel)
+                  .setTranscoder(transcoder)
+                  .setShouldOptimize(true)
+                  .setOpQueueMaxBlockTime(config.getTimeout())
+                  .setOpTimeout(config.getTimeout())
+                  .setReadBufferSize(config.getReadBufferSize())
+                  .setOpQueueFactory(opQueueFactory)
+                  .setEnableMetrics(MetricType.PERFORMANCE)
+                  .setWriteOpQueueFactory(opQueueFactory)
+                  .setReadOpQueueFactory(opQueueFactory)
+                  .setMetricCollector(
+                      new MetricCollector()
+                      {
+                        @Override
+                        public void addCounter(String name)
+                        {
+                          counters.put(name, new AtomicLong(0L));
+
+                          if(log.isDebugEnabled()) {
+                            log.debug("Add Counter [%s]", name);
+                          }
+                        }
+
+                        @Override
+                        public void removeCounter(String name)
+                        {
+                          counters.remove(name);
+
+                          if(log.isDebugEnabled()) {
+                            log.debug("Remove Counter [%s]", name);
+                          }
+                        }
+
+                        @Override
+                        public void incrementCounter(String name)
+                        {
+                          AtomicLong counter = counters.get(name);
+                          if (counter == null) {
+                            counters.putIfAbsent(name, new AtomicLong(0));
+                            counter = counters.get(name);
+                          }
+                          counter.incrementAndGet();
+
+                          if(log.isDebugEnabled()) {
+                            log.debug("Incrament [%s]", name);
+                          }
+                        }
+
+                        @Override
+                        public void incrementCounter(String name, int amount)
+                        {
+                          AtomicLong counter = counters.get(name);
+                          if (counter == null) {
+                            counters.putIfAbsent(name, new AtomicLong(0));
+                            counter = counters.get(name);
+                          }
+                          counter.addAndGet(amount);
+
+                          if(log.isDebugEnabled()) {
+                            log.debug("Increment [%s] %d", name, amount);
+                          }
+                        }
+
+                        @Override
+                        public void decrementCounter(String name)
```

Comment:
There's no future proof way to do it. The strings are hardcoded constants in the classes that report them.